src/main/cli/cli.c

```diff
@@ -4091,7 +4091,21 @@ STATIC_UNIT_TESTED void cliSet(char *cmdline)
                     valueChanged = true;
 
                     break;
-
+                case MODE_STRING: {
+                        const unsigned int len = strlen(cmdline);
+                        const uint8_t min = val->config.string.minlength;
+                        const uint8_t max = val->config.string.maxlength;
+                        const bool updatable = ((val->config.string.flags & STRING_FLAGS_WRITEONCE) == 0 || strlen((char *)cliGetValuePointer(val)) == 0);
```

Comment:
Not sure if introducing a special case for `board_name` (and probably more special syntax to also cover `manufacturer_id` and `signature`) makes sense here - these are treated as a special case pretty much everywhere else.

Reply:
more a thinking out loud... i am not sure we should really be worrying too much about the write  once ... people will get around it regardless.

Reply:
This is not so much intended as a 'lock down', more as a measure to make Unified Targets equivalent to legacy targets, by being able to detect and apply the correct Unified Target configuration even after the board configuration has been reset after a firmware upgrade.